WalletWasabi.Fluent/Controls/TagsBox.xaml.cs

```diff
@@ -0,0 +1,354 @@
+
+			CompositeDisposable.Dispose();
+			base.OnDetachedFromVisualTree(e);
+		}
+
+		private void CheckIsInputEnabled()
+		{
+			if (Items is IList x && ItemCountLimit > 0)
+			{
+				_isInputEnabled = x.Count < ItemCountLimit;
+			}
+		}
+
+		private void OnAutoCompleteBoxLostFocus(object? _, RoutedEventArgs e)
+		{
+			PseudoClasses.Set(":focus", false);
+		}
+
+		private void OnOnAutoCompleteBoxGotFocus(object? _, GotFocusEventArgs e)
+		{
+			PseudoClasses.Set(":focus", true);
+		}
+
+		private void OnTextInput(object? sender, TextInputEventArgs e)
+		{
+			if (!(sender is AutoCompleteBox autoCompleteBox))
+			{
+				return;
+			}
+
+			if (!_isInputEnabled)
+			{
+				e.Handled = true;
+				return;
+			}
+
+			if (RestrictInputToSuggestions &&
+				Suggestions is IList<string> suggestions &&
+				!suggestions.Any(x =>
+					x.StartsWith(autoCompleteBox.SearchText ?? "", true, CultureInfo.CurrentCulture)))
+			{
+				e.Handled = true;
+			}
+		}
+
+		protected override void UpdateDataValidation<T>(AvaloniaProperty<T> property, BindingValue<T> value)
+		{
+			if (property == ItemsProperty)
+			{
+				DataValidationErrors.SetError(this, value.Error);
+			}
+		}
+
+		protected override void OnGotFocus(GotFocusEventArgs e)
+		{
+			base.OnGotFocus(e);
+			FocusChanged(HasFocus());
+		}
+
+		protected override void OnLostFocus(RoutedEventArgs e)
+		{
+			base.OnLostFocus(e);
+			FocusChanged(HasFocus());
+		}
+
+		private bool HasFocus()
+		{
+			IVisual? focused = FocusManager.Instance.Current;
+
+			while (focused != null)
+			{
+				if (ReferenceEquals(focused, this))
+				{
+					return true;
+				}
+
+				// This helps deal with popups that may not be in the same
+				// visual tree
+				IVisual parent = focused.GetVisualParent();
+				if (parent is null)
+				{
+					// Try the logical parent.
+					if (focused is IControl element)
+					{
+						parent = element.Parent;
+					}
+				}
+
+				focused = parent;
+			}
+
+			return false;
+		}
+
+		private void FocusChanged(bool hasFocus)
+		{
+			// The OnGotFocus & OnLostFocus are asynchronously called and cannot
+			// reliably tell you that have the focus. All they do is let you
+			// know that the focus changed sometime in the past. To determine
+			// if you currently have the focus you need to do consult the
+			// FocusManager (see HasFocus()).
+
+			var wasFocused = _isFocused;
+			_isFocused = hasFocus;
+
+			if (hasFocus)
+			{
+				if (!wasFocused)
+				{
+					_autoCompleteBox?.Focus();
+				}
+			}
+
+			PseudoClasses.Set(":focus", hasFocus);
+			_isFocused = hasFocus;
+		}
+
+		private void OnDropDownClosed(object? sender, EventArgs e)
+		{
+			if (!(sender is AutoCompleteBox autoCompleteBox))
+			{
+				return;
+			}
+
+			var currentText = autoCompleteBox.Text ?? "";
+
+			if (currentText.Length == 0 ||
+				!(autoCompleteBox.SelectedItem is string selItem) ||
+				selItem.Length == 0 ||
+				currentText != selItem)
+			{
+				return;
+			}
+
+			AddTag(currentText.Trim());
```

Comment:
this should be renamed "SelectTag" or SelectItem?... you should select Item[0] from the suggestions list... rather than trying to get a matching string?